src/Domain/Creators/CylindricalBinaryCompactObject.hpp

```diff
@@ -163,15 +180,19 @@ class CylindricalBinaryCompactObject : public DomainCreator<3> {
   };
 
   struct InitialRefinement {
-    using type = size_t;
-    static constexpr Options::String help = {
-        "Initial refinement level. Applied to each dimension."};
-  };
-  struct AdditionToOuterRadialRefinementLevel {
-    using type = size_t;
+    using type =
+        std::variant<size_t, std::array<size_t, 3>,
+                     std::vector<std::array<size_t, 3>>,
+                     std::unordered_map<std::string, std::array<size_t, 3>>>;
     static constexpr Options::String help = {
-        "Addition to radial refinement level in the outer layer (CA and CB "
-        "blocks), beyond the refinement level set by InitialRefinement."};
+        "Initial refinement level. Specify one of: a single number, a list "
+        "representing [r, theta, perp], or such a list for every block in the "
+        "domain. Here 'r' is the radial direction normal to the inner and outer"
+        "boundaries, 'theta' is the periodic direction, and 'perp' is the third"
+        "direction. Note that for filled cylinders, 'r' is the 'z' direction "
+        "of the cylinder and 'perp' is the 'rho' direction of the cylinder, "
+        "but for cylindrical shells 'r' is the 'rho' direction of the cylinder "
+        "and 'perp' is the 'z' direction."};
   };
   struct InitialGridPoints {
```

Comment:
I'd suggest to expand this option over the blocks as well, since you're pretty much getting it "for free" with the setup you already have. Up to you though.

Reply:
Done.